src/control/server/mgmt_svc.go

```diff
@@ -23,6 +25,39 @@ import (
 	"github.com/daos-stack/daos/src/control/system/raft"
 )
 
+type (
+	batchRequest struct {
+		msg    proto.Message
+		ctx    context.Context
```

Comment:
As noted in the link, it's guidance, not a rule. And the reasons for it make sense -- passing a context as an argument encourages good design that allows for easy context scoping and reasoning about a system. In this case, the reason we are storing a context is for one purpose: To avoid a hang in the batch handler that is trying to send a response to a gRPC request that has been canceled ([code](https://github.com/daos-stack/daos/pull/11893/files#diff-c7ceef5c2c3cbaa758519a519c7c7cd76a3209b7d466b296a95db792749df421R55)).

We can't pass the gRPC request's context as a standalone argument because we're sending the request via a channel to the batch processing loop, so the only option we have is to add it to the `batchRequest` type, which can be thought of as a structured argument. Doing it that way is in keeping with the spirit of the guidance, in that we're not really storing the context in a long-lived struct, we're just using the struct to convey the context over a channel with some other related information.